docs/usage/graphql.md

```diff
@@ -2,11 +2,21 @@
 
 Once OAuth is complete, we can use the library's `GraphqlClient` to make requests to the GraphQL Admin API in a similar way. To do that, create an instance of `GraphqlClient` using the current shop URL and session `accessToken` in your app's endpoint.
 
+The `GraphQLClient`'s main method is `query`, which accepts a `GraphQLParams` object as its argument. `GraphQLParams` only requires the `data` parameter, but also optionally accepts `query`, `extraHeaders`, and `tries`:
+
+| Parameter      | Type                                | Notes                                                                                 | Required? | Default Value |
+| -------------- | ----------------------------------- | ------------------------------------------------------------------------------------- | :-------: | :-----------: |
+| `data`         | `Record<string, unknown> \| string` | Takes in either the query as a string, and an object containing a query and variables |   True    |     none      |
+| `query`        | `Record<string, string \| number>`  | An optional query object to be appended to the request                                |   False   |     none      |
+| `extraHeaders` | `Record<string, string \| number>`  | Any additional headers you want to send with your request                             |   False   |     none      |
+| `tries`        | `number`                            | The maximum number of times to retry the request _(must be >= 0)_                     |   False   |      `1`      |
+
 ```ts
-  const session = await Shopify.Utils.loadCurrentSession(req, res); // load the current session to get the `accessToken`
-  const client = new Shopify.Clients.Graphql(session.shop, session.accessToken); // GraphQLClient accepts the same arguments as RestClient
-  const products = await client.query({ // use client.query and pass your query as `data`
-    data: `{
+const session = await Shopify.Utils.loadCurrentSession(req, res); // load the current session to get the `accessToken`
```

Comment:
I don't think we have a set rule about this (and perhaps we should define one!) but I'd like to consider the ergonomics of using trailing comments in example code, as opposed to putting the comment on the line above, especially with longer lines:

```js
// comment
code
```
vs:
```js
code // comment
```

Dunno if anyone else has preferences/thoughts about this.

Reply:
Personally, I prefer to put comments before the code, it can be harder to see which line the comment applies to, but it makes lines shorter and keep things visually aligned.

Reply:
I think there's a lot of debate about this in "actual" code as well. On the one hand, I see the inline/trailing comment as being more clearly tied to that line, or one of its arguments/behavior etc, but the above-line comments might be better for readability and line-length. My feeling on inline comments tends to depend on their overall length, I think. But having a set rule/convention for this would be great!

Reply:
@paulomarg I'm fine with before the code and proper sentence casing as well. Probably better from a presentational standpoint.